spec/lib/bookmarks_bulk_action_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+RSpec.describe BookmarksBulkAction do
+  fab!(:user) { Fabricate(:user, refresh_auto_groups: true) }
+  fab!(:bookmark_1) { Fabricate(:bookmark) }
+  fab!(:bookmark_2) { Fabricate(:bookmark) }
+
+  describe "delete" do
+    describe "when user can't delete" do
+      it "does NOT delete the bookmarks" do
+        Guardian.any_instance.stubs(:can_edit?).returns(false)
+        Guardian.any_instance.stubs(:can_delete?).returns(false)
```

Comment:
Here too we should also have a few tests with the bookmark edit permission.

Reply:
I checked the [bookmark guardian](https://github.com/discourse/discourse/blob/main/lib/guardian/bookmark_guardian.rb) implementation, and it seems that currently, all users who can edit can also delete. So, the tests will be the same. I still added them to avoid future regressions, though.